### ISSUE TITLE
bind_cols error due to mismatch between # rows in results-tibble (level = class) and total # of grid cells

### ISSUE THREAD

**Issue by gponce-ars** [r-spatialecology/landscapemetrics#175]
Hi, 

I'm just trying to plot the corresponding raster of one metric for one specific class. 
This is how I'm trying to get this:

**Note:** The error I'm getting here is happening only in 1 case out of +60

```
g <- dt_rasters$g_list[[17]] # A grid created with st_make_grid
r <- dt_rasters$r_list[[17]] # A raster, integer values (1-4)

plot(r, col=c('khaki4', 'darkgreen', 'orange', 'black'))
plot(g, add = TRUE)
```

![Screen Shot 2020-04-24 at 5 27 05 PM](https://user-images.githubusercontent.com/20215136/80266465-d8248880-8650-11ea-81d6-7024105eb431.png)

Then I just run sample_lsm() by class level.
```
list_results <-sample_lsm(r, g, level = 'landscape', 
                                            metric =c('lpi','pland'), 
                                            return_raster = TRUE)

# Now, just want to plot the metric (`lpi`) raster of one of the classes:  1
res<- list_results  %>% filter((is.na(class) | class == 1) & metric == 'lpi')

```

This is where the error happens... I searched for similar error but those I found were using 'landscape' and I need information by class.

```
# Using the grid and tibble `res` we should be able to get the raster 
raster1 <- bind_cols(g,res)
Error: Argument 2 must be length 63, not 62
```
That difference of 1 is caused because class 2(green) is the only class present in the cell highlighted below (red circle), therefore, class 1 has one cell less than the total number of grid cells (NA cells + cells with data). 

![Screen Shot 2020-04-24 at 5 32 51 PM](https://user-images.githubusercontent.com/20215136/80266616-a6f88800-8651-11ea-98a1-0c255cf09c8a.png)

In summary, when something like this happens, the number of rows in the tibble differs between classes and those from the grid used.  Is there any way to create NA tibble rows for the cases when there's no pixel of class X present within a grid cell, the same way is done with fully empty cells.

Hopefully, this is clear enough... Maybe I'm missing a parameter that can be set to force this?

G

**Comment by mhesselbarth** [comment]
Hey, 

I am not sure if I understood your problem correctly, but I think this is related to #170 

Does the following code using the `complete()` function from the `tidyr` package fix your problem? 
 
``` r
library(dplyr)
library(landscapemetrics)
library(raster)
library(tidyr)

lc_example <- rep(x = c(1, 2), each = 50) %>% 
  matrix(nrow = 10, ncol = 10, byrow = FALSE) %>% 
  raster(xmn = 0, xmx = 100, ymn = 0, ymx = 100)

sample_points <- matrix(c(25, 50, 75, 50), ncol = 2, byrow = TRUE)

plot(lc_example)
points(x = sample_points[,1], y = sample_points[,2])
```

![](https://i.imgur.com/MZuvGrg.png)

``` r

result <- landscapemetrics::sample_lsm(lc_example, y = sample_points, 
                                       size = 5, what = c("lsm_c_ca", "lsm_c_pland"))
#> Warning: Please use 'check_landscape()' to ensure the input data is valid.

col_order <- names(result)

result <- tidyr::complete(result, class, tidyr::nesting(layer, level, id, metric,
                                                        plot_id, percentage_inside)) %>% 
  dplyr::arrange(layer, plot_id, level, metric, class, id)

result <- result[, col_order]

result
#> # A tibble: 8 x 8
#>   layer level class    id metric  value plot_id percentage_inside
#>   <int> <chr> <int> <int> <chr>   <dbl>   <int>             <dbl>
#> 1     1 class     1    NA ca       0.01       1               100
#> 2     1 class     2    NA ca      NA          1               100
#> 3     1 class     1    NA pland  100          1               100
#> 4     1 class     2    NA pland   NA          1               100
#> 5     1 class     1    NA ca      NA          2               100
#> 6     1 class     2    NA ca       0.01       2               100
#> 7     1 class     1    NA pland   NA          2               100
#> 8     1 class     2    NA pland  100          2               100
```

<sup>Created on 2020-04-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
```

**Comment by gponce-ars** [comment]
Thanks, @mhesselbarth 

Yes. That `tidyr::complete -> nesting -> arrange` produced the structure needed with the missing NA's.

And yes, it is totally related to this [#170 ](https://github.com/r-spatialecology/landscapemetrics/issues/170).

Cheers!
Guillermo
